backend/battling/forms.py

```diff
@@ -33,47 +33,47 @@ def __init__(self, *args, **kwargs):
         self.fields["creator"].widget = forms.HiddenInput()
 
     def clean_opponent(self):
-        opponent_email = self.cleaned_data["opponent"]
-        try:
-            opponent = User.objects.get(email=opponent_email)
-        except User.DoesNotExist:
-            self.is_guest = True
-            opponent = User.objects.create(email=opponent_email)
-            random_password = get_random_string(length=64)
-            opponent.set_password(random_password)
-            opponent.save()
+        if "opponent" in self.cleaned_data:
+            opponent_email = self.cleaned_data["opponent"]
+            try:
+                opponent = User.objects.get(email=opponent_email)
+            except User.DoesNotExist:
+                self.is_guest = True
+                opponent = User.objects.create(email=opponent_email)
+                random_password = get_random_string(length=64)
+                opponent.set_password(random_password)
+                opponent.save()
         return opponent
```

Comment:
This will throw a `UnboundLocalError` because the variable is not defined on `else`. Please test if you can get this error.

Reply:
I actually tried to put an else, but when I ran the pre-commit, it told me that I should remove the else because it wasn't needed. I didn't get any errors, though